x-pack/plugins/security_solution/docs/adrs/threat_hunting/0001_saving_of_timeline_associated_saved_objects.md

```diff
@@ -0,0 +1,87 @@
+
+## Considered Options
+
+- Auto-saving timeline when an a note or pinned event is created
+- Disabling notes and pinned events until the timeline is saved
+- Opening the save modal when clicking the note/pin buttons on an unsaved timeline
+- Caching notes and pinned events locally, saving when timeline is saved
+
+## Decision Outcome
+
+Chosen option: **TBD**
+
+### Confirmation
+
+_PR link will be added later_
+
+## Discussion of options
+
+### Auto-saving timeline when an a note or pinned event is created
+
+In this proposed solution we're bringing back parts of the "auto-save" behaviour. A request to save the associated saved object will create a timeline if the current one hasn't been saved. The request then returns the `timelineId` and `timelineVersion` which will be stored the active timeline.
+This approach has been implemented in https://github.com/elastic/kibana/pull/178212 .
+
+| Pros                                         | Cons                                                                        |
+| -------------------------------------------- | --------------------------------------------------------------------------- |
+| Mimics the previous auto-save behaviour      | Timelines that are created this way, will all be called `Untitled Timeline` |
+| Easy to test in unit and in acceptance tests | Might surprise users, now that timelines are not autosaved anymore          |
+| Simple, very few things can go wrong         |                                                                             |
+| No public API changes necessary              |                                                                             |
+
+### Disabling notes and pinned events until the timeline is saved
+
+Disabling all note/pin buttons and lists until the timeline is saved makes sure that no associated saved objects can be created with a missing `timelineId`. Subsequently, the code that creates timelines when `timelineId` is missing should be removed as well.
+This approach has been implemented in https://github.com/elastic/kibana/pull/178525 .
+
+| Pros                                         | Cons                                                                      |
+| -------------------------------------------- | ------------------------------------------------------------------------- |
+| No `Untitled timeline`s                      | Requires public API changes                                               |
+| Easy to test in unit and in acceptance tests | Users might use note/pin less often, since they are disabled "by default" |
+| Simple, very few things can go wrong         |                                                                           |
```

Comment:
One more con for this was that it might lead to orphan notes/pinned events when timelines are duplicated. or created from a rule that has investigation guide as mentioned here: https://github.com/elastic/kibana/pull/178525#pullrequestreview-1954211840

Reply:
Thanks, added it here: https://github.com/elastic/kibana/pull/179830/commits/793d461e9a6d624fde8d7e5bc6e863554272b9bd